source/common/router/upstream_request.cc

```diff
@@ -248,12 +465,40 @@ void UpstreamRequest::acceptHeadersFromRouter(bool end_stream) {
   router_sent_end_stream_ = end_stream;
 
   conn_pool_->newStream(this);
```

Comment:
Why don't we delay constructing a new stream from the connection pool until the upstream filter chain has been traversed? i.e. move this to the end of `acceptHeadersFromRouter`?

Reply:
generally many streams are blocking, so kicking off the stream early would be important to many users.  I think we can make it configurable in the long run though.

Reply:
I think this is fine, but might be worth a comment here on the implication of this. I have a vague worry that this might cause issues for idle timeouts or something if there are filters that pause the request flow for a while. I guess in general all upstream timeouts would be impacted due to filters so probably not a big deal, but might be worth a comment/TODO here either way.